src/main/java/rx/subjects/ReplaySubject.java

```diff
@@ -115,13 +117,13 @@ public void call(SubjectObserver<T> o) {
                 try {
                     for (;;) {
                         int idx = o.<Integer>index();
-                        int sidx = state.index;
+                        int sidx = state.index.get();
```

Comment:
`state` is a hidden field in the anonymous inner class so it is subject to the same re-read problems.